library/src/main/java/com/novoda/downloadmanager/DownloadBatch.java

```diff
@@ -136,7 +140,7 @@ private long getTotalSize(List<DownloadFile> downloadFiles) {
     }
 
     void pause() {
-        if (downloadBatchStatus.status() == PAUSED) {
+        if (downloadBatchStatus.status() == PAUSED || downloadBatchStatus.status() == DOWNLOADED) {
```

Comment:
This prevents pause being called when the asset is already `DOWNLOADED`. The same for `resume` below.

Reply:
Yeah good one, this solves lots of issues when the library is used without checking current download statuses, as pausing a paused download

Reply:
Extract `downloadBrachStatus.status()` to a variable 😅  

Also applicable for `resume()` method.

Reply:
3d12073